code/modules/antagonists/changeling/powers/spiders.dm

```diff
@@ -1,18 +1,145 @@
+	gibbed = TRUE
+	return ..()
+
+/mob/living/simple_animal/hostile/poison/giant_spider/hunter/infestation_spider/death()
+	if(!gibbed)
+		gib()
+		return
+	var/mob/owner_mob = locateUID(owner_UID)
+	var/datum/action/changeling/spiders/S = locate() in owner_mob.actions
+	S.spider_counter--
+	return ..()
+
+/mob/living/simple_animal/hostile/poison/giant_spider/hunter/infestation_spider/examine(mob/user)
+	. = ..()
+	if(user.UID() == owner_UID)
+		. += "<span class='notice'>It is currently angy!</span>"
+
+/mob/living/simple_animal/hostile/poison/giant_spider/hunter/infestation_spider/AltShiftClick(mob/user)
+	. = ..()
+	if(user.UID() == owner_UID)
+		spider_order(user)
+
+/mob/living/simple_animal/hostile/poison/giant_spider/hunter/infestation_spider/proc/spider_order(mob/user)
+	switch(currentOrder)
+		if(0)
+			to_chat(user, "<span class='notice'>We order the giant spider to follow us but attack anything on sight.</span>")
+			currentOrder = 1
+		if(1)
+			to_chat(user, "<span class='notice'>We order the giant spider to follow us and stay calm, only attacking if we or it are attacked.</span>")
+			currentOrder = 2
+		if(2)
+			to_chat(user, "<span class='notice'>We order the giant spider to remain idle and calm, only attacking if we or it are attacked.</span>")
+			currentOrder = 3
+		if(3)
+			to_chat(user, "<span class='notice'>We order the giant spider to remain idle but attack anything on sight.</span>")
+			currentOrder = 0
+	handle_automated_movement()
+
+/mob/living/simple_animal/hostile/poison/giant_spider/hunter/infestation_spider/handle_automated_movement() //Hacky and ugly.
+	. = ..()
+	var/list/around = view(src, vision_range)
+	switch(currentOrder)
+		if(0)
+			Find_Enemies()
+			walk(src,0)
+			return TRUE
+		if(1)
+			Find_Enemies()
+			if(!busy)
+				for(var/mob/living/carbon/C in around)
+					if(faction_check_mob(C))
+						if(Adjacent(C))
+							return TRUE
+						//stop_automated_movement = TRUE
+						Goto(C, 0.5 SECONDS, 1)
+						//stop_automated_movement = FALSE
+				return TRUE
+		if(2)
+			if(!busy)
+				for(var/mob/living/carbon/C in around)
+					if(faction_check_mob(C))
+						if(Adjacent(C))
+							return TRUE
+						//stop_automated_movement = TRUE
+						Goto(C, 0.5 SECONDS, 1)
+						//stop_automated_movement = FALSE
+				//walk(src,0)
+				return TRUE
+		if(3)
+			walk(src,0)
+			return TRUE
+
+	for(var/mob/living/simple_animal/hostile/poison/giant_spider/hunter/infestation_spider/H in around)
+		if(faction_check_mob(H) && !attack_same && !H.attack_same)
+			H.enemies |= enemies
+
+/mob/living/simple_animal/hostile/poison/giant_spider/hunter/infestation_spider/ListTargets()
+	if(!enemies.len)
+		return list()
+	var/list/see = ..()
+	see &= enemies // Remove all entries that aren't in enemies
+	return see
+
+// Aaaaaaaaaaaaaaaaaaaaaaa
+
+/mob/living/simple_animal/hostile/poison/giant_spider/hunter/infestation_spider/proc/Find_Enemies()
+	var/list/around = view(src, vision_range)
```

Comment:
This should be passed from `handle_automated_movement()` so we don't have to call it twice.

Reply:
👍 Handled